scripts/machine_translation/train_transformer.py

```diff
@@ -0,0 +1,448 @@
+    else:
+        raise NotImplementedError
+    # TODO(sxjscience) Try to use the new DataLoader in https://github.com/apache/incubator-mxnet/pull/17841
+    batchify_fn = bf.Tuple(bf.Pad(), bf.Pad(), bf.Stack(), bf.Stack(), bf.Stack())
+    # TODO(sxjscience) Support auto-bucket-size tuning
+    train_batch_sampler = FixedBucketSampler(lengths=[(ele[2], ele[3]) for ele in data_train],
+                                             batch_size=args.batch_size,
+                                             num_buckets=args.num_buckets,
+                                             ratio=args.bucket_ratio,
+                                             shuffle=True,
+                                             use_average_length=True,
+                                             bucket_scheme=bucket_scheme,
+                                             seed=args.seed)
+    train_data_loader = gluon.data.DataLoader(data_train, batch_sampler=train_batch_sampler,
+                                              batchify_fn=batchify_fn,
+                                              num_workers=0)
+    logging.info(train_batch_sampler)
+    val_data_loader = gluon.data.DataLoader(data_val, batch_size=16, batchify_fn=batchify_fn,
+                                            num_workers=0, shuffle=False)
+    num_batches = len(train_data_loader)
+    for v in model.collect_params().values():
+        if v.grad_req != 'null':
+            v.grad_req = 'add'
+    model.collect_params().zero_grad()
+    model_averager = AverageSGDTracker(model.collect_params())
+    log_start_time = time.time()
+    num_params, num_fixed_params = None, None
+    # TODO(sxjscience) Add a log metric class
+    accum_count = 0
+    loss_denom = 0
+    n_train_iters = 0
+    log_wc = 0
+    log_avg_loss = 0.0
+    log_loss_denom = 0
+    for epoch_id in range(args.epochs):
+        n_epoch_train_iters = 0
+        sample_data_l = []
+        for batch_id, samples in enumerate(train_data_loader):
+            # TODO(sxjscience)
+            #  Here, there are two ways to load the data in the multi-GPU setting.
+            #   1. Iterate the data loader for NGPU times to obtain NGPU minibatches and
+            #      calculate/aggregate the gradient of each batch
+            #   2. Load the data with a larger batch size (NGPU * batch_size). Then, use the
+            #      split_load functionality in Gluon to split the minibatch to multiple GPUs.
+            #  Since we need to pad the data, we use the first strategy in the implementation.
+            #  However, when Ragged is enabled, we can revise the code to use the second strategy.
+            sample_data_l.append(samples)
+            if len(sample_data_l) == len(ctx_l) or batch_id == num_batches - 1:
+                # Accumulate the gradient
+                loss_l = []
+                accum_count += 1
+                for sample_data, ctx in zip(sample_data_l, ctx_l[:len(sample_data_l)]):
```

Comment:
So we do not use data parallel?

Reply:
Updated the number. May need help for tuning.